### ISSUE TITLE
Separate file I/O from analysis

### ISSUE THREAD

**Issue by bobturneruk** [Mesnage-Org/Mass-Spec-MS1-Analysis#5]
This function currently takes strings and then loads files and does analysis.

https://github.com/Dr-Encore/Mass-Spec-MS1-Analysis/blob/c2735e8fcf73575d9c7d701f729a465331fefcc9/matching%20v7.py#L190

It should take objects representing the file contents in-memory (data frames?) so that it can be called in different ways. Refactor something like this pseudo-nonsense:

```
def load_database(database_file_name:string) -> db_table:data_frame

def load_csv(csv_file_name:string) -> csv_table:data_frame

def main (db_table:data_frame, csv_table:data_frame) -> output:data_frame

def save_output(output:data_frame, filename:string):
    #save to disk
```

**Comment by bobturneruk** [comment]
![image](https://user-images.githubusercontent.com/34244196/100454348-9fe89b00-30b4-11eb-9f03-790864fc9db1.png)

**Comment by bobturneruk** [comment]
Closed by #26
